docs/content/preview/tutorials/AI/ai-llamaindex-openai.md

```diff
@@ -0,0 +1,250 @@
+image: /images/tutorials/ai/icons/llamaindex-icon.svg
+headcontent: Use YugaybteDB as the database backend for RAG applications
+menu:
+  preview:
+    identifier: tutorials-ai-llamaindex-openai
+    parent: tutorials-ai
+    weight: 60
+type: docs
+---
+
+This tutorial demonstrates how use LlamaIndex to build RAG (Retrieval-Augmented Generation) applications. By using the LlamaIndex [SQLJoinQueryEngine](https://docs.llamaindex.ai/en/stable/examples/query_engine/SQLJoinQueryEngine.html), the application can query a [PostgreSQL-compatible](https://www.yugabyte.com/postgresql/postgresql-compatibility/) YugabyteDB database from natural language. It can then infer whether to query a secondary vector index to fetch documents. In this case, the secondary index contains the Wikipedia pages of S&P 500 companies.
+
+## Prerequisites
+* Install Python3
+* Install Docker
+
+## Set up the application
+Download the application and provide settings specific to your deployment:
+
+1. Clone the repository.
+
+    ```sh
+    git clone https://github.com/YugabyteDB-Samples/yugabytedb-llamaindex-sp500-search.git
+    ```
+
+2. Install the application dependencies.
+
+    Dependencies can be installed in a virtual environment, or globally on your machine.
+
+    * Option 1 (recommended): Install Dependencies from *requirements.txt* in virtual environment
+        ```sh
+        python3 -m venv yb-llamaindex-env
+        source yb-llamaindex-env/bin/activate
+        pip install -r requirements.txt
+        # NOTE: Users with M1 Mac machines should use requirements-m1.txt instead:
+        # pip install -r requirements-m1.txt
+        ```
+
+    * Option 2: Install Dependencies Globally
+        ```sh
+        pip install llama-index
+        pip install psycopg2
+        # NOTE: Users with M1 Mac machines should install the psycopg2 binary instead:
+        # pip install psycopg2-binary
+        pip install python-dotenv
+        ```
+3. Create an [OpenAI API Key](https://platform.openai.com/api-keys) and store it's value in a secure location. This will be used to connect the application to the LLM to generate SQL queries, infer results and generate the proper response.
+
+4. Configure the application environment variables in `{project_directory/.env}`.
+
+## Get Started with YugabyteDB
+
+YugabyteDB is a [PostgreSQL-compatible](https://www.yugabyte.com/postgresql/postgresql-compatibility/) distributed database.  
+
+Start a 3-node YugabyteDB cluster in Docker (or feel free to use another deployment option):
+
+```sh
+# NOTE: if the ~/yb_docker_data already exists on your machine, delete and re-create it
+mkdir ~/yb_docker_data
+
+docker network create custom-network
+
+docker run -d --name yugabytedb-node1 --net custom-network \
+    -p 15433:15433 -p 7001:7000 -p 9001:9000 -p 5433:5433 \
+    -v ~/yb_docker_data/node1:/home/yugabyte/yb_data --restart unless-stopped \
+    yugabytedb/yugabyte:2.20.1.0-b97 \
```

Comment:
let's use the "latest" version everywhere

Reply:
I believe @ddhodge updates this version programmatically based on the version of the docs the user is viewing. I.e. 
```yugabytedb/yugabyte:{{< yb-version version="preview" format="build">}} ```

@ddhodge should we always use this syntax?